### PR TITLE
Add package-lock updater

### PR DESCRIPTION
**Pull request by Scy-D2L**
- Adds [update-package-lock](https://github.com/BrightspaceUI/actions/tree/main/update-package-lock#update-package-lockjson-action) workflow to automatically update dependencies if they pass CI.
- Schedules said workflow to run on Mondays an hour before dependabot for minimum conflict

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1 +1,3 @@
 *       @BrightspaceUILabs/nimbus
+golden/
```
Comment (tdevos-d2l) [inline .github/CODEOWNERS:2]: Just curious, what does adding the `golden` folder here do? Does this not add us as reviews when only goldens are being updated?

Comment (Scy-D2L) [reply .github/CODEOWNERS:2]: Correct, this is actually unrelated to the package updater but I forgot to note it down - the general purpose is actually for updating vdiffs - it should keep the PRs from trying to auto-add the team if someone marks them as ready (we have the same codeowner config in CE). The person who opened the PR is usually the only one we need pinged for those.

Comment (tdevos-d2l) [reply .github/CODEOWNERS:2]: Ah, makes sense!

**Review by tdevos-d2l — APPROVED** [review]

**Review by jeroach — APPROVED** [review]

**Comment by tdevos-d2l** [comment]
Could we do the same for annotation-viewer? :)

**Pull request merged**